pkg/operator/operator_test.go

```diff
@@ -0,0 +1,131 @@
+package operator
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/certificates/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+const dir = "_tmp"
+
+func TestEnsureCerts(t *testing.T) {
+	var (
+		timeout     = 3 * time.Second
+		ctx, cancel = context.WithTimeout(context.Background(), timeout)
+		csr         *v1.CertificateSigningRequest
+		fqdn        = fmt.Sprintf("system:node:%s.%s.svc", NameOperator, "")
+		certBytes   = []byte{1, 2, 3, 4}
+	)
+	t.Cleanup(cancel)
+
+	os.Mkdir(dir, 0777)
+	defer func() {
+		os.RemoveAll(dir)
+	}()
```

Comment:
You can simply write `defer os.RemoveAll(dir)`. The additional closure would only help if you changed the value of `dir` again further down as it will then only be resolved when the deferred function is actually called on exit.